### PR TITLE
add remove command

### LINKED ISSUE #69: Add remove command
Add `remove` command to remove specific items from HomeworkList, AttendanceList and GradeProgressList.

### PR DESCRIPTION
**Pull request by farisanadia**
i implemented it the same way edit is implemented but idk if that's good
feel free to review thoroughly idrk whats going on

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/RemoveCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        model.setFullView();
+    }
+
+    public void setDayView() {
+        model.setDayView();
+    }
+
+    public void setListView() {
+        model.setListView();
+    }
```
Comment (AshiqurRah) [inline src/test/java/seedu/address/logic/commands/RemoveCommandTest.java:35]: One other test that you could include could be execute_noFieldSpecifiedUnfilteredList_success, which is to check whether removeCommand will return error message when there is no field specified.

**Review by AshiqurRah — COMMENTED** [review]
Looks good so far, lets also include the RemoveCommandParserTest tooo

File: src/main/java/seedu/address/logic/parser/RemoveCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                ArgumentTokenizer.tokenize(args, PREFIX_HOMEWORK,
+                        PREFIX_GRADEPROGRESS, PREFIX_ATTENDANCE, PREFIX_SESSION);
+
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemoveCommand.MESSAGE_USAGE), pe);
+        }
```
Comment (cfyjoanna) [inline src/main/java/seedu/address/logic/parser/RemoveCommandParser.java]: there isn't supposed to be a person index for the remove command

Comment (farisanadia) [reply src/main/java/seedu/address/logic/parser/RemoveCommandParser.java]: the edit command has a person index though

Comment (cfyjoanna) [reply src/main/java/seedu/address/logic/parser/RemoveCommandParser.java]: yea that's why it's my job to update it LOL

Comment (farisanadia) [reply src/main/java/seedu/address/logic/parser/RemoveCommandParser.java]: oh LOL
pls update mine too thx

Comment (farisanadia) [reply src/main/java/seedu/address/logic/parser/RemoveCommandParser.java]: that was  a jk btw

Comment (cfyjoanna) [reply src/main/java/seedu/address/logic/parser/RemoveCommandParser.java]: u will be cursed with a CI that never passes

File: src/test/java/seedu/address/logic/commands/RemoveCommandTest.java (hunk 1 above)
Comment (farisanadia) [reply src/test/java/seedu/address/logic/commands/RemoveCommandTest.java:35]: yessir i will try my best

File: src/test/java/seedu/address/logic/parser/RemoveCommandParserTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        // zero index
+        assertParseFailure(parser, "0", MESSAGE_INVALID_FORMAT);
+
+        // invalid arguments being parsed as preamble
+        assertParseFailure(parser, "1 some random string", MESSAGE_INVALID_FORMAT);
+
+        // invalid prefix being parsed as preamble
+        assertParseFailure(parser, "1 i/ string", MESSAGE_INVALID_FORMAT);
+    }
+
```
Comment (AshiqurRah) [inline src/test/java/seedu/address/logic/parser/RemoveCommandParserTest.java:39]: RemoveCommandParserTest looks good, just one additional small test of successful parse would do

**Review by AshiqurRah — COMMENTED** [review]
Just one small part, other than that LGTM!

File: src/test/java/seedu/address/logic/parser/RemoveCommandParserTest.java (hunk 3 above)
Comment (farisanadia) [reply src/test/java/seedu/address/logic/parser/RemoveCommandParserTest.java:39]: OKIE dokie

**Review by cfyjoanna — APPROVED** [review]
LGTM can add more tests in future 🟡

**Pull request merged**